### PR TITLE
slackcat 0.7 (new formula)

### PR DESCRIPTION
**Pull request by bcicen**

### CODE REVIEW

File: Library/Formula/slackcat.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Slackcat < Formula
+  desc "Simple command-line Utility to post snippets to Slack."
+  homepage "https://github.com/vektorlab/slackcat"
+  url "https://github.com/vektorlab/slackcat/archive/v0.7.tar.gz"
+  version "0.7"
```
Comment (DomT4) [inline Library/Formula/slackcat.rb]: This line won't be required. We can read simple-ish versions automatically.

File: Library/Formula/slackcat.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  sha256 "b51ab794af2a0014b5372944699d7ff9c88af4e1860206abee3a0f9bbc70d147"
+
+  depends_on "go"
+
+  def install
+    platform = `uname`.downcase.strip
```
Comment (DomT4) [inline Library/Formula/slackcat.rb]: This will just be `Darwin` here.

File: Library/Formula/slackcat.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class Slackcat < Formula
+  desc "Simple command-line Utility to post snippets to Slack."
+  homepage "https://github.com/vektorlab/slackcat"
+  url "https://github.com/vektorlab/slackcat/archive/v0.7.tar.gz"
+  version "0.7"
+  sha256 "b51ab794af2a0014b5372944699d7ff9c88af4e1860206abee3a0f9bbc70d147"
+
+  depends_on "go"
```
Comment (DomT4) [inline Library/Formula/slackcat.rb]: Presumably not at runtime, only during the build?

File: Library/Formula/slackcat.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  def install
+    platform = `uname`.downcase.strip
+
+    unless ENV["GOPATH"]
```
Comment (DomT4) [inline Library/Formula/slackcat.rb]: Please use our established system for handling go. You can find an example [here](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/gdm.rb#L18-L34)

File: Library/Formula/slackcat.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      ENV["GOPATH"] = "/tmp"
+    end
+
+    system "make"
+    bin.install "build/slackcat-0.7-#{platform}-amd64" => "slackcat"
+
+    puts "Ready to go! Generate a new Slack key with 'slackcat --configure'"
```
Comment (DomT4) [inline Library/Formula/slackcat.rb]: Please don't `puts` in formula.

**Comment by bcicen** [comment]
Only remaining test failure I see now is repo age.

**Comment by dunn** [comment]
Also the Makefile runs `go get`, which we don't allow: https://github.com/vektorlab/slackcat/blob/master/Makefile#L6

We'll need to add the dependencies as `go_resources` and run `go build` directly.  See [`hugo.rb`](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/hugo.rb) for an example.

File: Library/Formula/slackcat.rb (hunk 4 above)
Comment (DomT4) [inline Library/Formula/slackcat.rb]: Ping on this comment. It's still relevant, re `go_resource`s and setting `GOPATH`.

File: Library/Formula/slackcat.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+    mkdir_p buildpath/"src/github.com/vektorlab/"
+    ln_sf buildpath, buildpath/"src/github.com/vektorlab/slackcat"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "-ldflags", "-s -X main.version="version"", "-o", bin/"slackcat", "slackcat.go"
+  end
+
+  test do
+    assert_match /slackcat version #{version}/, shell_output("#{bin}/slackcat -v").chomp
```
Comment (mikemcquaid) [inline Library/Formula/slackcat.rb]: The `.chomp` is unneeded here.

File: Library/Formula/slackcat.rb [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  def install
+    ENV["GOBIN"] = bin
+    ENV["GOPATH"] = buildpath
+    ENV["GOHOME"] = buildpath
+
+    mkdir_p buildpath/"src/github.com/vektorlab/"
+    ln_sf buildpath, buildpath/"src/github.com/vektorlab/slackcat"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    system "go", "build", "-ldflags", "-s -X main.version="version"", "-o", bin/"slackcat", "slackcat.go"
```
Comment (mikemcquaid) [inline Library/Formula/slackcat.rb]: `"-s -X main.version=#{version}"` is a bit nicer.

**Comment by bcicen** [comment]
Thanks for the comments and pointing me in the right direction. I've made the mentioned changes to the formula.

**Comment by bcicen** [comment]
Squashed and pushed commits fixing formula

**Comment by bfontaine** [comment]
Thank you for your contribution to Homebrew @bcicen; we appreciate it! :tada:

**Pull request closed without merging**